contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,277 @@
+
+  /**
+   * @type {Array.<Object.<string,*>>}
+   * @private
+   */
+  this.datasources_ = datasources;
+
+  /**
+   * @type {TypeaheadOptions}
+   * @export
+   */
+  this.options = /** @type {TypeaheadOptions} */ ({
+    highlight: true,
+    hint: undefined,
+    minLength: undefined
+  });
+
+  /**
+   * @type {Array.<TypeaheadDataset>}
+   * @export
+   */
+  this.datasets = [];
+  for (var i = 0; i < this.datasources_.length; i++) {
+    var datasource = this.datasources_[i];
+    var groupByProperties = datasource['groupByProperties'] || [];
+    var filter;
+    do {
+      var dataTitle = datasource['dataTitle'];
+
+      //Add an optional filter function to keep objects only from one
+      //"layername" from a GMF's fulltextsearch service.
+      if (groupByProperties[0]) {
+        filter = this.filterLayername_(groupByProperties[0]);
+        dataTitle = dataTitle + groupByProperties[0];
+        groupByProperties.shift();
+      } else {
+        filter = undefined;
+      }
+
+      this.datasets.push(this.createDataset_({
+        'dataTitle': dataTitle,
+        'filter': filter,
+        'projection': datasource['projection'],
+        'url': datasource['url']
+      }));
+
+    } while (groupByProperties.length > 0);
+  }
+
+
+
+  /**
+   * @type {ngeox.SearchDirectiveListeners}
+   * @export
+   */
+  this.listeners = /** @type {ngeox.SearchDirectiveListeners} */ ({
+    selected: goog.bind(gmf.SearchController.selected_, this)
+  });
+
+};
+
+
+/**
+ * @return {ol.layer.Vector} The vector layer.
+ * @private
+ */
+gmf.SearchController.prototype.createVectorLayer_ = function() {
+  var vectorLayer = new ol.layer.Vector({
+    source: new ol.source.Vector({
+      useSpatialIndex: false
+    })
+  });
+  // Use vectorLayer.setMap(map) rather than map.addLayer(vectorLayer). This
+  // makes the vector layer "unmanaged", meaning that it is always on top.
+  vectorLayer.setMap(this.map_);
+  return vectorLayer;
+};
+
+
+/**
+ * @param {Object} config the config of te dataset
+ * @return {TypeaheadDataset} a typeahead dataset.
+ * @private
+ */
+gmf.SearchController.prototype.createDataset_ = function(config) {
+  var directive_scope = this.scope_;
+  var compile = this.compile_;
+  var bloodhoundEngine = this.createAndInitBloodhound_(config);
+  return /** @type {TypeaheadDataset} */ ({
+    source: bloodhoundEngine.ttAdapter(),
+    display: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: /* TypeaheadTemplates */ ({
+      header: function() {
+        return '<div class="header">' + config['dataTitle'] + '</div>';
+      },
+      suggestion: function(suggestion) {
+        var feature = /** @type {ol.Feature} */ (suggestion);
+
+        var scope = directive_scope.$new(true);
+        scope['feature'] = feature;
+        scope['click'] = function(event) {
+          window.alert(feature.get('label'));
```

Comment:
I've not removed that. I'm not sur if I must remove all the "button" part of this "suggestion" function or if I must use a "cleaner" fonction to make an alert. I don't know if we already have something to create an alert..?